base/float.jl

```diff
@@ -414,9 +422,121 @@ muladd(x::T, y::T, z::T) where {T<:IEEEFloat} = muladd_float(x, y, z)
 # TODO: faster floating point fld?
 # TODO: faster floating point mod?
 
-rem(x::T, y::T) where {T<:IEEEFloat} = rem_float(x, y)
+function unbiased_exponent(x::T) where {T<:IEEEFloat}
+    exp = (reinterpret(Unsigned, x) & exponent_mask(T)) >> mantissa_width(T)
+    return exp
+end
+
+function explicit_mantissa_noinfnan(x::T) where {T<:IEEEFloat}
+    subnormal = !issubnormal(x) ? significand_mask(T) + uinttype(T)(1) : uinttype(T)(0)
+    return subnormal | (significand_mask(T) & reinterpret(Unsigned, x))
```

Comment:
```suggestion
    m = mantissa(x)
    issubnormal(x) || (m |= significand_mask(T) + uinttype(T)(1))
    return m
```
I personally find the logic easier to follow this way.